### PR TITLE
Implement backend endpoint for search by instructor

### LINKED ISSUE #9: FEATURE INSTRUCTOR - Implement backend
Implement the backend for instructor-based course search (Issue #25 on proj-courses).

- [x] Coverage testing with jacoco passes
- [x] Mutation testing with pitest passes
- [x] GitHub CI is green
- [x] Functionality works for queries on localhost with Swagger
  - [x] You can select a start quarter, end quarter, and input an instructor name
  - [x] If you enter a partial name, e.g. `WANG`, all versions of Wang are returned (e.g. `WANG R K`, `WANG JIAJUN`, etc)
  - [x] The search should _not_ be case-sensitive
  - [x] If you enter a complete name such as `WANG R K` you should get only courses with that name precisely
  - [x] If a professor teaches in multiple departments, their courses from all departments will show up
- [x] Functionality works for queries on Dokku qa with Swagger
  - [x] You can select a start quarter, end quarter, and input an instructor name
  - [x] If you enter a partial name, e.g. `WANG`, all versions of Wang are returned (e.g. `WANG R K`, `WANG JIAJUN`, etc)
  - [x] The search should _not_ be case-sensitive
  - [x] If you enter a complete name such as `WANG R K` you should get only courses with that name precisely
  - [x] If a professor teaches in multiple departments, their courses from all departments will show up

### PR DESCRIPTION
**Pull request by wq12li**
In this PR, I've added backend support for searching courses by instructor and a quarter range. For the time being, this implementation is relatively simple and does not consider the difference between lectures and sections.

My implementation imitates that of the CourseOverTimeController.java file, and is mostly the same.

### CODE REVIEW

**Review by khtran1 — DISMISSED** [review]
LGTM

File: src/main/java/edu/ucsb/cs156/courses/controllers/CourseOverTimeInstructorController.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    @Autowired
+    ConvertedSectionCollection convertedSectionCollection;
+
+    @ApiOperation(value = "Get a list of courses over time, filtered by instructor name")
+    @GetMapping(value = "/instructorsearch", produces = "application/json")
+    public ResponseEntity<String> search(
+        @RequestParam String startQtr,
+        @RequestParam String endQtr,
+        @RequestParam String instructor
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/courses/controllers/CourseOverTimeInstructorController.java]: See my feedback on PR #19 here: 
https://github.com/ucsb-cs156-s23/proj-courses-s23-7pm-3/pull/19#discussion_r1208653470

I think adding similar `@ApiParam` annotations will make this easier for future programmers to work with.

**Review by pconrad — CHANGES_REQUESTED** [review]
Looks good... I'll be testing on the staff qa site before giving a final approval.

I would like you to add the `@ApiParam` annotation to the parameters to the new API endpoints though.

**Review by khtran1 — APPROVED** [review]
LGTM

**Review by pconrad — APPROVED** [review]
LGTM!   Will test on staff-qa before merging.

**Pull request merged**